### PR TITLE
SSCISCI-677: stop NPE when language is null

### PR DESCRIPTION
**Pull request by benouaer**
### JIRA link (if applicable) ###
- https://tools.hmcts.net/jira/browse/SSCSCI-677

### Change description ###
- stop NPE when language is null

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by hashimalisolirius — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/job/DataMigrationJob.java [hunk 1]
```diff
@@ -39,7 +40,12 @@ public boolean readyToRun() {
     public void process() {
         String languageColumn = isRollback ? EXISTING_LANGUAGE_COLUMN : MAPPED_LANGUAGE_COLUMN;
         log.info("Processing Interpreter data {} job", isRollback ? "rollback" : "migration");
-        migrationService.process(languageColumn);
+        try {
+            migrationService.process(languageColumn);
+        } catch (IOException e) {
+            log.error("{} job failed to decode encodedDataString", isRollback ? "rollback" : "migration", e);
```
Comment (nitinprabhuhmcts) [inline src/main/java/uk/gov/hmcts/reform/sscs/job/DataMigrationJob.java:46]: can we please have a test to cover this ?

File: src/main/java/uk/gov/hmcts/reform/sscs/services/ccd/CcdCasesSender.java [hunk 2]
```diff
@@ -71,28 +72,31 @@ public void sendUpdateCcdCases(SscsCaseData gapsCaseData, SscsCaseDetails existi
...
+            var caseData = sscsCcdConvertService.getCaseData(startEventResponse.getCaseDetails().getData());
+
+            var isInExcludedState = startEventResponse.getCaseDetails().getState().equals(VOID_STATE.toString())
+                || startEventResponse.getCaseDetails().getState().equals(DORMANT_APPEAL_STATE.toString());
+            var needInterpreter = YesNo.YES.getValue()
+                .equals(caseData.getAppeal().getHearingOptions().getLanguageInterpreter());
+            var languageAlreadySet = language.equals(caseData.getAppeal().getHearingOptions().getLanguages());
+
+            if (isInExcludedState || !needInterpreter || languageAlreadySet) {
+                log.info("Skipping case ({}) as language value already matching ({})", caseId, language);
```
Comment (nitinprabhuhmcts) [inline src/main/java/uk/gov/hmcts/reform/sscs/services/ccd/CcdCasesSender.java:86]: or in excluded state so worth logging state as well

File: src/main/java/uk/gov/hmcts/reform/sscs/util/MigrationDataEncoderApp.java [hunk 3]
```diff
@@ -42,24 +37,29 @@ public static void main(String[] args) {
...
 
         } catch (IOException e) {
             log.error("There was a problem encoding the migration data, migration file ({}), encoded string file ({})",
-                MIGRATION_FILE, ENCODED_STRING_FILE);
+                MIGRATION_FILE, ENCODED_STR_FILE);
             throw new RuntimeException(e);
         }
     }
+
+    public static String compressAndB64Encode(String text) throws IOException {
```
Comment (nitinprabhuhmcts) [inline src/main/java/uk/gov/hmcts/reform/sscs/util/MigrationDataEncoderApp.java:58]: not sure this method is covered by test

File: src/main/java/uk/gov/hmcts/reform/sscs/job/DataMigrationJob.java (hunk 1 above)
Comment (benouaer) [reply src/main/java/uk/gov/hmcts/reform/sscs/job/DataMigrationJob.java:46]: there is one

File: src/main/java/uk/gov/hmcts/reform/sscs/util/MigrationDataEncoderApp.java (hunk 3 above)
Comment (benouaer) [reply src/main/java/uk/gov/hmcts/reform/sscs/util/MigrationDataEncoderApp.java:58]: I haven't tested this class at all, as I've considered a utility that doesn't actually get run during the migration. we run it separately beforehand.
I can test it if required

**Review by nitinprabhuhmcts — APPROVED** [review]

**Pull request merged**